app/livechat/server/lib/Helper.js

```diff
@@ -205,30 +222,29 @@ export const forwardRoomToDepartment = async (room, guest, departmentId) => {
 	// Fake the department to forward the inquiry - Case the forward process does not success
 	// the inquiry will stay in the same original department
 	inquiry.department = departmentId;
-	room = await RoutingManager.delegateInquiry(inquiry);
-	if (!room) {
+	const roomTaken = await RoutingManager.delegateInquiry(inquiry);
+	if (!roomTaken) {
 		return false;
 	}
 
-	const { servedBy } = room;
-	// if there was an agent assigned to the chat and there is no new agent assigned
-	// or the new agent is not the same, then the fowarding process successed
-	if (oldServedBy && (!servedBy || oldServedBy._id !== servedBy._id)) {
-		Subscriptions.removeByRoomIdAndUserId(rid, oldServedBy._id);
-		Messages.createUserLeaveWithRoomIdAndUser(rid, { _id: oldServedBy._id, username: oldServedBy.username });
-		LivechatRooms.changeDepartmentIdByRoomId(rid, departmentId);
-		LivechatInquiry.changeDepartmentIdByRoomId(rid, departmentId);
-		// Update the visitor's department
-		const { token } = guest;
-		Livechat.setDepartmentForGuest({ token, department: departmentId });
-
-		if (servedBy) {
-			const { _id, username } = servedBy;
-			Messages.createUserJoinWithRoomIdAndUser(rid, { _id, username });
-		}
+	const { servedBy } = roomTaken;
+	if (oldServedBy && servedBy && oldServedBy._id === servedBy._id) {
+		return false;
+	}
+
+	if (oldServedBy) {
+		removeAgentFromSubscription(rid, oldServedBy);
+	}
 
-		return true;
+	if (servedBy) {
+		Messages.createUserJoinWithRoomIdAndUser(rid, servedBy);
 	}
 
-	return false;
+	LivechatRooms.changeDepartmentIdByRoomId(rid, departmentId);
+	LivechatInquiry.changeDepartmentIdByRoomId(rid, departmentId);
+
+	const { token } = guest;
+	Livechat.setDepartmentForGuest({ token, department: departmentId });
+
+	return true;
```

Comment:
very good refactor here as well.. code is much easier to understand, congrats :clap: